src/cdk/table/can-stick.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** @docs-private */
+export type Constructor<T> = new(...args: any[]) => T;
+
+/**
+ * Interface for a mixin to provide a directive with a function that checks if the sticky input has
+ * been changed since the last time the function was called. Essentially adds a dirty-check to the
+ * sticky value.
+ * @docs-private
+ */
+export interface CanStick {
+  /** Whether the sticky input has changed since it was last checked. */
+  _hasStickyChanged: boolean;
+
+  /** Whether the sticky value has changed since this was last called. */
+  hasStickyChanged(): boolean;
+
+  /** Resets the dirty check for cases where the sticky state has been used without checking. */
+  resetStickyChanged(): void;
+}
+
+/**
+ * Mixin to provide a directive with a function that checks if the sticky input has been
+ * changed since the last time the function was called. Essentially adds a dirty-check to the
+ * sticky value.
```

Comment:
Include something in the description here that explains why the actual `sticky` property isn't included in the mixin